src/devices_models/devices/common/cost_functions.jl

```diff
@@ -339,6 +347,16 @@ function ps_cost(
     # If array is full of tuples with zeros return 0.0
     all(iszero.(last.(cost_array))) && return JuMP.AffExpr(0.0)
     variable = get_variable(psi_container, var_name)[index, :]
+    settings_ext = get_ext(get_settings(psi_container))
+    if haskey(settings_ext, "variable_on")
+        var_name = settings_ext["variable_on"]
+        bin = get_variable(psi_container, var_name)[index, :]
+    elseif haskey(settings_ext, "parameter_on")
+        param_name = settings_ext["parameter_on"]
+        bin = get_parameter_container(psi_container, param_name).parameter_array[index, :]
+    else
+        bin = nothing
+    end
```

Comment:
I don't follow what you are trying to do here. Can you please elaborate.

Reply:
For the SOS2 implementation you need to know if there is variable or a parameter for Status(ON). This depends on the formulation and feedforward. The variable/paramter is used in this constraint ` sum( x_pwl(i) for i in breakpoints) == varon(t)/param_on(t)`. This wasn't needed before as one of the breakpoint was (0,0).